### ISSUE TITLE
Adding an `invoke_from_event_loop()` with non-`Send` closure

### ISSUE THREAD

**Issue by Enyium** [slint-ui/slint#5988]
I use Rust.

Because I need to receive the `WM_DISPLAYCHANGE` message from Windows, I create a window class and window using [this crate](https://docs.rs/windows-helpers/latest/windows_helpers/win32_app/window/struct.WindowClass.html) of mine. I create an invisible window that allows me to receive broadcast messages from the OS. When I receive the message, I need to update the Slint window.

I can't be sure whether this can cause problems with Slint (it didn't yet). The code runs in the main thread, but Slint won't be aware of it. If I have an `Rc<RefCell<T>>` or a similar non-`Send` type on whose inner type I need to call a method, directly doing so in the extra OS window's window procedure needs to be **documented** [here](https://docs.rs/slint/latest/slint/fn.invoke_from_event_loop.html) as safe with Slint. In the case that this isn't true, but additionally because of possible reentrancy problems - depending on what WinAPI calls the user makes - an **`invoke_from_event_loop()` equivalent** that doesn't force you away from a non-`Send` type like `Rc<RefCell<T>>` would come in handy, so possibly problematic calls can be deferred without awareness of other threads than the main thread (also with regard to the interior mutability Slint uses internally). The name could perhaps be `defer_until_event_loop()`.

**Comment by ogoffart** [comment]
I believe slint::spawn_local should do what you need.
Otherwise you can use the send_wrapper crate

**Comment by Enyium** [comment]
It works. Must this be viewed as kind of a sledgehammer approach? Because I'm not actually using async features, just synchronous code, but now need an `async move { ... }` block.

I'll assume that I can rest assured that this `Future` will be able to be executed with any Cargo feature set that allows to build a Windows GUI app.

---

What I mentioned in the initial comment and this usage should ideally still be clearly documented for `slint::invoke_from_event_loop()` and `slint::spawn_local()`.

**Comment by ogoffart** [comment]
I wanted to add "see also slint::spawn_local" from the invoke_from_event_loop function, but it is unfortunately not possible to make a link because invoke_from_event_loop is defined in i-slint-core while spawn_local is defined in api/rs/slint, which is a different crate.

**Comment by Enyium** [comment]
I have made the experience that links are sometimes rendered differently locally with `cargo doc` and on docs.rs - meaning, links may break or work when they locally behaved otherwise. If both functions are reexported by the `slint` crate, I don't see why the links shouldn't work on docs.rs. Of course, I can't guarantee it.

I think it was first and foremost relative links that suddenly behaved differently for me on docs.rs, so that I had to revise them.
